src/java.desktop/unix/classes/sun/awt/X11/XCanvasPeer.java

```diff
@@ -62,36 +62,33 @@ public GraphicsConfiguration getAppropriateGraphicsConfiguration(
         if (graphicsConfig == null || gc == null) {
             return gc;
         }
-        // Opt: Only need to do if we're not using the default GC
 
-        int screenNum = ((X11GraphicsDevice)gc.getDevice()).getScreen();
+        final X11GraphicsDevice newDev = getSameScreenDevice(gc);
+        final int visualToLookFor = graphicsConfig.getVisual();
 
-        X11GraphicsConfig parentgc;
-        // save vis id of current gc
-        int visual = graphicsConfig.getVisual();
-
-        X11GraphicsDevice newDev = (X11GraphicsDevice) GraphicsEnvironment.
-            getLocalGraphicsEnvironment().
-            getScreenDevices()[screenNum];
-
-        for (int i = 0; i < newDev.getNumConfigs(screenNum); i++) {
-            if (visual == newDev.getConfigVisualId(i, screenNum)) {
-                // use that
-                graphicsConfig = (X11GraphicsConfig)newDev.getConfigurations()[i];
-                break;
+        final GraphicsConfiguration[] configurations = newDev.getConfigurations();
+        for (final GraphicsConfiguration config : configurations) {
+            final X11GraphicsConfig x11gc = (X11GraphicsConfig) config;
+            if (visualToLookFor == x11gc.getVisual()) {
+                graphicsConfig = x11gc;
             }
         }
-        // just in case...
-        if (graphicsConfig == null) {
-            graphicsConfig = (X11GraphicsConfig) GraphicsEnvironment.
-                getLocalGraphicsEnvironment().
-                getScreenDevices()[screenNum].
-                getDefaultConfiguration();
-        }
```

Comment:
why did you get rid of the just in case ?
Actually why is this method being modified at all ? 
I mean besides adding locking ..

Reply:
> why did you get rid of the just in case ?

I honestly couldn't find any way for this code to get executed. I can keep it there, but then I would have to explain why it is needed.

> Actually why is this method being modified at all ?
The changes are really semantically null. All I did was shorten a 30 lines method to 15 and gave a name to a piece of it (but that's in the next comment).

Reply:
@prrace polite ping